File: frontend/src/types/portfolio.ts
export interface Position {
  isin: string;
  current_quantity: number;
  average_cost: number;
  total_invested: number;
  realized_pnl: number;
  unrealized_pnl: number;
  market_value: number;
  last_calculated_at: string;
}

export interface Portfolio {
  client_id: string;
  positions: Position[];
  total_value: number;
  total_realized_pnl: number;
  total_unrealized_pnl: number;
}

File: backend/app/schemas/portfolio.py
"""Portfolio response schemas."""

from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel


class PositionOut(BaseModel):
    isin: str
    current_quantity: Decimal
    average_cost: Decimal
    total_invested: Decimal
    realized_pnl: Decimal
    unrealized_pnl: Decimal
    market_value: Decimal
    last_calculated_at: datetime

    model_config = {"from_attributes": True}


class PortfolioOut(BaseModel):
    client_id: str
    positions: list[PositionOut]
    total_value: Decimal
    total_realized_pnl: Decimal
    total_unrealized_pnl: Decimal
